src/Server/Workspace.php

```diff
@@ -216,28 +216,55 @@ public function xdependencies(): array
     }
 
     /**
-     * @param Options|null $settings
+     * Fires when client changes settings in the client
+     *
+     * The default paramter type is Options but it also accepts different types
+     * which will be transformed on demand.
+     *
+     * Currently only the vscode format is supported
+     *
+     * @param mixed|null $settings
+     * @return bool
+     * @throws \Exception Settings format not valid
      */
-    public function didChangeConfiguration(Options $settings = null)
+    public function didChangeConfiguration($settings = null): bool
     {
+        // List of options that affect the indexer
+        $indexerOptions = ['fileTypes'];
+
         if ($settings === null) {
-            return;
+            return false;
+        }
+
+        // VSC sends the settings with the config section as main key
+        if ($settings instanceof \stdClass && $settings->phpIntelliSense) {
+            $mapper = new \JsonMapper();
+            $settings = $mapper->map($settings->phpIntelliSense, new Options);
+        }
+
+        if (!($settings instanceof Options)) {
```

Comment:
Do you have to tell him to that?
Because for me JsonMapper never threw an exception when the schema didn't match.
It silently ignored invalid properties and used the defaults in the `Options`.